src/dune_rules/merlin/ocaml_index.ml

```diff
@@ -0,0 +1,120 @@
+open Import
+
+let ocaml_index sctx ~dir =
+  Super_context.resolve_program ~loc:None ~dir sctx "ocaml-index"
+;;
+
+let index_path_in_obj_dir obj_dir =
+  let dir = Obj_dir.obj_dir obj_dir in
+  Path.Build.relative dir "cctx.ocaml-index"
+;;
+
+let project_index ~build_dir = Path.Build.relative build_dir "project.ocaml-index"
+
+let cctx_rules cctx =
+  let open Memo.O in
+  (* Indexing is performed by the external binary [ocaml-index] which performs
+     full shape reduction to compute the actual definition of all the elements in
+     the typedtree. This step is therefore dependent on all the cmts of those
+     definitions are used by all the cmts of modules in this cctx. *)
+  let sctx = Compilation_context.super_context cctx in
+  let* additional_libs =
+    let open Resolve.Memo.O in
+    let+ non_compile_libs =
+      (* The indexer relies on the load_path of cmt files. When
+         [implicit_transitive_deps] is set to [false] some necessary paths will
+         be missing.These are passed to the indexer with the `-I` flag.
+
+         The implicit transitive libs correspond to the set:
+         (requires_link \ req_compile) *)
+      let* req_link = Compilation_context.requires_link cctx in
+      let+ req_compile = Compilation_context.requires_compile cctx in
+      List.filter req_link ~f:(fun l -> not (List.exists req_compile ~f:(Lib.equal l)))
+    in
+    Lib_flags.L.include_flags non_compile_libs (Lib_mode.Ocaml Byte)
+  in
+  let dir = Compilation_context.dir cctx in
+  let aggregate =
+    let obj_dir = Compilation_context.obj_dir cctx in
+    let fn = index_path_in_obj_dir obj_dir in
+    let includes =
+      Resolve.peek additional_libs |> Result.value ~default:Command.Args.empty
```

Comment:
Is the peek here intentional? In case we can't resolve one of the libs, we just pretend all of them don't exist without letting the user know?

Reply:
That's a good catch, I removed the peek in [4c2aee3](https://github.com/ocaml/dune/pull/10422/commits/4c2aee34457b9c4e2ec78dc7f405e748a5558add)